apps/system/js/bluetooth.js

```diff
@@ -216,6 +221,67 @@ var Bluetooth = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   * Since BTv1 get onenabled event before onadapteradded event,
+   * We need watch if adapter is retrieved.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    return new Promise((resolve) => {
+      if (this.defaultAdapter !== null) {
+        resolve(this.defaultAdapter);
+      }
+
+      this.watch('defaultAdapter', (id, oldval, newval) => {
```

Comment:
Fred, I ready the MDN(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/watch).

And the doc. mention:
`Warning: Generally you should avoid using watch() and unwatch() when possible. These two methods are implemented only in Gecko, and they're intended primarily for debugging use. In addition, using watchpoints has a serious negative impact on performance, which is especially true when used on global objects, such as window. You can usually use setters and getters or proxies instead. See Browser compatibility for details. Also, do not confuse Object.watch with Object.observe.`

So, I'm wondering that it makes worst performance in our system app. Could you find out other solve here? We cannot jump into the risk since the doc. is high light first.